wordpress-mobile/WordPress-iOS
# Issues/4035 update readerpost model

Refs: #4035 
This PR: 
- Adds some new fields to the `ReaderPost` model
- removes some old obsolete fields and methods 
- updates the remote service to populate the new fields
- adds some tests for newly added remote service methods

Since we're still in a dev cycle I didn't bump the model revision for these changes. 

To test: 
- Since the new fields are not yet used confirm the app launches and the reader syncs and displays its contents without error.
- Run the test suite and confirm there are no errors

@SergioEstevao would you be up for a quick review? 

Needs Review: @SergioEstevao
